### PR TITLE
refactor: Organize compactor component responsibilities

### PR DESCRIPTION
**Pull request by carols10cents**
Closes https://github.com/influxdata/idpe/issues/17273.

What started out as a small refactoring became a larger set of cleanups as I got to know the code-- I'm hoping this fits well with the changes to compaction rules that Joe and Nga are working on. If not, please let me know and I'll undo/redo!

I highly recommend reviewing commit-by-commit: each commit should have a description of the change in the commit message, and each commit compiles and passes the compactor tests. I tried to make each commit a fairly straightforward change.

### CODE REVIEW

**Review by domodwyer — APPROVED** [review]
I had a quick look through and overall it's a fantastic cleanup ❤️  I am violent agreement with some of those renames!

Given the lack of functional changes, and existing tests that (should be) validating correctness, I propose merging this ASAP as it is going to be a bit of a conflict magnet!

**Review by Joe-Blount — APPROVED** [review]
earlier this week I was thinking there were extra layers of functions around executing the plans, and now its gone :) 
LGTM.
I think it will cause conflicts with what I'm working on, but probably for the better.

**Review by NGA-TRAN — APPROVED** [review]
Very nice refactor. I have gone over commit by commit. I at first paused a bit with new names. They are more general than the old ones but putting them in the abstract context we have created, new names make more sense. I have suggested to add comments to help future coder to understand better with this abstraction.

I can see this refactor at least resolve a couple of ticket (refactor and metrics/logs)

Thanks @carols10cents  ❤️

**Comment by NGA-TRAN** [comment]
BTW, the fact that we have a massive set of tests and none of them failed mean this is a real refactor.

@carols10cents : The tests that I think we still miss is scratchpad. If you know it better now, maybe a ticket to add tests for it. This is the only concern I am afraid when we refactor something like this, we will change the resource allocation/release in the scratchpad by mistake

File: compactor2/src/components/post_classification_partition_filter/logging.rs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        partition_info: &PartitionInfo,
+        files: &[ParquetFile],
+    ) -> Result<bool, DynError> {
+        let res = self.inner.apply(partition_info, files).await;
+        match &res {
+            Ok(true) => {
+                debug!(
+                    partition_id = partition_info.partition_id.get(),
+                    filter_type = self.filter_type,
+                    "NOT filtered partition"
```
[the remark this answers is not in the document]
Comment (carols10cents) [reply compactor2/src/components/post_classification_partition_filter/logging.rs:56]: I made it match the filters that happen before classification: https://github.com/influxdata/influxdb_iox/blob/e9290531093be4ad644d9915083b2ca434dcdb85/compactor2/src/components/partition_filter/logging.rs#L51

File: compactor2/src/components/post_classification_partition_filter/mod.rs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+pub mod metrics;
+pub mod mock;
+pub mod possible_progress;
+
+/// Filters partition based on ID and parquet files after the files have been classified.
+///
+/// May return an error. In this case, the partition will be marked as "skipped".
+#[async_trait]
+pub trait PostClassificationPartitionFilter: Debug + Display + Send + Sync {
+    /// Return `true` if the if the compactor should run a
```
[the remark this answers is not in the document]
Comment (carols10cents) [reply compactor2/src/components/post_classification_partition_filter/mod.rs:17]: Great catch! I'm adding a commit that fixes this spot and the same problem over here too: https://github.com/influxdata/influxdb_iox/blob/e9290531093be4ad644d9915083b2ca434dcdb85/compactor2/src/components/partition_filter/mod.rs#L31

**Pull request merged**